extmod/asyncio/stream.py

```diff
@@ -146,14 +175,24 @@ async def _serve(self, s, cb):
             except:
                 # Ignore a failed accept
                 continue
+            if s2:
+                if ssl and _ssl:
+                    if isinstance(ssl, _ssl.SSLContext):
```

Comment:
I don't think it's necessary to have all these checks.  Just do simply `if ssl:`.